### ISSUE TITLE
Error updated v 1.0.0

### ISSUE THREAD

**Issue by liamlain** [humbertogontijo/homeassistant-roborock#309]
Here the log attached 

This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/roborock/__init__.py:141
Integration: Roborock (documentation, issues)
First occurred: 4:26:54 PM (1 occurrences)
Last logged: 4:26:54 PM

Error setting up entry xxxxxx@gmail.com for roborock: Device 6Nzd68UCacQefAxPR0tvcg not found among {}
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/roborock/__init__.py", line 92, in async_setup_entry
    devices_network = await get_local_devices_info(_device)
  File "/config/custom_components/roborock/__init__.py", line 141, in get_local_devices_info
    raise ConfigEntryError(f"Device {device.duid} not found among {devices_network}")
homeassistant.exceptions.ConfigEntryError: Device 6Nzd68UCacQefAxPR0tvcg not found among {}

**Comment by q1t1** [comment]
Same here, I have the following error :

Retrying setup: Timeout after 4 seconds waiting for response

 Logger: roborock.api
Source: runner.py:179
First occurred: 13:12:14 (41 occurrences)
Last logged: 13:19:57

    Timeout after 4 seconds waiting for response
    [Errno 32] Broken pipe

**Comment by liamlain** [comment]
Thanks for the reply.
but i donno how to do that.anywhere i can a tutorial or doc about that?thank you

**Comment by humbertogontijo** [comment]
Basically your robot and HA instance needs to be in the same network

**Comment by liamlain** [comment]
Yes my roborock and my HA is under same network.
but my Roborock connected wifi 2.4ghz and my HA connected 5ghz.This is the reason ?

**Comment by humbertogontijo** [comment]
Hmm, unless you separated 2.4 and 5ghz into different sub lans, no

**Comment by humbertogontijo** [comment]
Can you start your vacuum and then try to reload the integration?

**Comment by liamlain** [comment]
i tested to disable 5ghz and only enable 2.4 thn restart HA ,the intergration is working now .
And also i re-enable back 5ghz. Roborock intergration stil work..im not sure is it 2.4 and 5ghz problem.
howevery my roborock is back now. 

btw i using tplink deco m9 mesh

**Comment by Eidolf** [comment]
Hello, I am in with the problem, but on my end it is clear if it needs an broadcast.
I devide my networks into VLAN's for SmartHome Devices and my office clients.
This scenario works flawlessly before the update.
Is there any special reason for the change?

But I have to say that I appreciate your previous versions very much and I was indescribably happy that someone took on this task.

**Comment by humbertogontijo** [comment]
@Eidolf the integration was using mqtt before. Now it's calling the device locally. It still uses mqtt, but only for the map

**Comment by humbertogontijo** [comment]
My idea is to let the ip be configured manually if listening for the broadcast do not work

**Comment by stevesreed** [comment]
Manual IP entry sounds great! Any ETA on that option?

I can use the roborock with HA at all now, I tried rolling back to 0.1.2, but can get that to work either now.

**Comment by polysa** [comment]
I have the same issue. Roborock and HA instance under the same vlan. HA can ping Roborock IP. I even tested to setup when Roborock is cleaning. 

Always get: Retrying setup: Timeout after 4 seconds waiting for response

**Comment by Xyaren** [comment]
Same error here.

My Homeassistant runs in Docker (NOT in host mode)
Could this be an issue ?

The IP is also in the APP, so available via some kind of api/mqtt.
Maybe it is worth trying that ip first before using the broadcast ?

**Comment by humbertogontijo** [comment]
@Xyaren what you mean by the ip is also in the app? There is a way to retrieve the ip using mqtt (get_network_info command)

**Comment by Xyaren** [comment]
Could this IP be used instead of sending a Broadcast ? 
The IP is reachable from within the container, but broadcasting is not due to the bridging.

**Comment by grahamsz** [comment]
> I devide my networks into VLAN's for SmartHome Devices and my office clients.

I have my wifi and wired segregated and ran into the same issue. I was able to configure my router to [relay the broadcast](https://github.com/britannic/ubnt-bcast-relay) packets on port 58866 and it immediately started working again.

**Comment by Eidolf** [comment]
> > I devide my networks into VLAN's for SmartHome Devices and my office clients.
> 
> I have my wifi and wired segregated and ran into the same issue. I was able to configure my router to [relay the broadcast](https://github.com/britannic/ubnt-bcast-relay) packets on port 58866 and it immediately started working again.

That is a very good information. I will wait on the next version from humbertogontijo, because I think he will get the job done and also it is easier to have all necassary things on one place.
But as a backup if the change is to complicated, I'll try it similarly.

**Comment by humbertogontijo** [comment]
Version 1.0.1 is out hopefully fixing the issue with unreachable broadcasts

**Comment by Eidolf** [comment]
@humbertogontijo I tried it out, but it doesn't help on my end. But I think it should help most of the guys with the problem.
The error change to "Failed to connect to (correct IP address)"
And thanks for taking care.

**Comment by humbertogontijo** [comment]
@Eidolf I'll add in a future release the option to use the cloud

**Comment by shred86** [comment]
For anyone with a similar setup, I upgraded from 0.1.2 to 1.0.1 without any issues. I run Home Assistant and my Roborock S7 MaxV Ultra on separate VLANs. I do not have any mDNS/Bonjour reflectors setup, but I do have a firewall rule that allows anything from the VLAN Home Assistant is on to connect to devices on the VLAN the Roborock S7 is on.

I also setup the integration again from a fresh HA install (dev container) and everything setup just fine. I wasn't completely clear on how the IP address for the vacuum was being obtained but looking at the source code, I think it's doing this:

1. Look for the broadcast coming from the Roborock vacuum. This obviously has the limitation of Home Assistant and the Roborock vacuum having to be on the same network subnet so the broadcast can be heard.
2. If the above doesn't work, it appears to be using the cloud API to get the network information for the Roborock vacuum.

Please correct me if that's incorrect but that appears to be the case based on what I mentioned above (HA and vacuum are on separate VLANs and I'm able to set it up with everything working). I think this is great and doesn't require having to setup some sort of mDNS reflector to repeat the broadcast across VLANs. Thanks for all the work @humbertogontijo!

**Comment by stevesreed** [comment]
When 1.0 I could not get thinks working (Roborock and HA on separate VLANs), so I rolled back to 0.1.2 and everything worked fine. 

When 1.0.1 came out, I updated again and this time everything worked perfectly. Thanks!

**Comment by humbertogontijo** [comment]
@shred86 you are absolutely right. If broadcast fails a request for the mqtt api is made to get the local ip

**Comment by Eidolf** [comment]
After reading the comment from @shred86 I tried it this morning again ( yesterday I make a couple of try's, also after changing firewall rules on my sophos SG) and now it works. So your fix helps also on my side.
Thanks alot for the fix @humbertogontijo

**Comment by Utopia69** [comment]
Update to 1.0.1 didn't solve for me:

Config entry 'XXX@gmail.com' for roborock integration not ready yet: id=22416 Timeout after 4 seconds; Retrying in background
Config entry 'XXX@gmail.com' for roborock integration not ready yet: id=37751 Timeout after 4 seconds; Retrying in background
Config entry 'XXX@gmail.com' for roborock integration not ready yet: unsupported operand type(s) for +: 'int' and 'list'; Retrying in background
Config entry 'XXX@gmail.com' for roborock integration not ready yet: id=74914 Timeout after 4 seconds; Retrying in background

**Comment by q1t1** [comment]
I got some errors after upgrading too.
Deleting and reconfiguring the intégration solved it.

**Comment by Utopia69** [comment]
> I got some errors after upgrading too. Deleting and reconfiguring the intégration solved it.

Delete & reconfigure didn't work for me :-(

**Comment by barrelltitor** [comment]
Updating to 1.0.1 made the device no longer work for me either

```
2023-05-13 14:23:45.628 INFO (MainThread) [roborock.local_api] Connecting to 192.168.88.245
2023-05-13 14:23:45.631 DEBUG (MainThread) [roborock.local_api] id=43484 Requesting method get_room_mapping with None
2023-05-13 14:23:45.633 WARNING (MainThread) [homeassistant.config_entries] Config entry 'xxxxx@xxxx.xxxx' for roborock integration not ready yet: Failed connecting to 192.168.88.245; Retrying in background
2023-05-13 14:23:45.639 DEBUG (MainThread) [roborock.local_api] id=96848 Requesting method get_status with None
2023-05-13 14:23:45.639 DEBUG (MainThread) [roborock.local_api] id=32794 Requesting method get_dnd_timer with None
2023-05-13 14:23:45.640 DEBUG (MainThread) [roborock.local_api] id=84646 Requesting method get_clean_summary with None
2023-05-13 14:23:45.640 DEBUG (MainThread) [roborock.local_api] id=67374 Requesting method get_consumable with None
2023-05-13 14:23:45.653 INFO (MainThread) [roborock.local_api] Connected to 192.168.88.245
2023-05-13 14:23:45.665 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.665 INFO (MainThread) [roborock.local_api] Connecting to 192.168.88.245
2023-05-13 14:23:45.667 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.668 INFO (MainThread) [roborock.api] Roborock client disconnected
2023-05-13 14:23:45.668 WARNING (MainThread) [roborock.api] [Errno 32] Broken pipe
2023-05-13 14:23:45.676 INFO (MainThread) [roborock.local_api] Connected to 192.168.88.245
2023-05-13 14:23:45.681 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.681 INFO (MainThread) [roborock.local_api] Connecting to 192.168.88.245
2023-05-13 14:23:45.689 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.690 INFO (MainThread) [roborock.api] Roborock client disconnected
2023-05-13 14:23:45.695 INFO (MainThread) [roborock.local_api] Connected to 192.168.88.245
2023-05-13 14:23:45.705 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.706 INFO (MainThread) [roborock.local_api] Connecting to 192.168.88.245
2023-05-13 14:23:45.711 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.711 INFO (MainThread) [roborock.api] Roborock client disconnected
2023-05-13 14:23:45.711 WARNING (MainThread) [roborock.api] [Errno 32] Broken pipe
2023-05-13 14:23:45.727 INFO (MainThread) [roborock.local_api] Connected to 192.168.88.245
2023-05-13 14:23:45.733 DEBUG (MainThread) [roborock.local_api] Disconnecting from 192.168.88.245
2023-05-13 14:23:45.734 INFO (MainThread) [roborock.api] Roborock client disconnected
```

My Home Assistant is in docker, but has host network access and nothing else has any problems. They're on the same subnet as well. It seems it decides to disconnect from the robot as soon as it manages to connect to it.

**Comment by Utopia69** [comment]
same here HA also in docker, with host system in same subnet as roborock.

**Comment by barrelltitor** [comment]
This morning I figured to try re-downloading again and restarting HA and it worked this time without having to reconfigure. Maybe removing and then re-adding the integration might fix this by getting rid of any leftovers

**Comment by Eidolf** [comment]
> This morning I figured to try re-downloading again and restarting HA and it worked this time without having to reconfigure. Maybe removing and then re-adding the integration might fix this by getting rid of any leftovers

This was in the end also my fix, but done that the evening before also a couple of times

**Comment by HabelStefan** [comment]
I still having the issue of the S8 becoming unavailable.
Getting a lot of errors from roborock.local_api.

Deinstalled and readding the integration did not help. 

Any ideas?

**Comment by Lash-L** [comment]
I believe this is fixed, if it is not, please ping me or open a new issue.
